eradiate/solvers/onedim/tests/test_app.py

```diff
@@ -132,67 +104,81 @@ def test_onedim_solver_app_run():
     assert set(results["lo"].dims) == {"sza", "saa", "vza", "vaa", "wavelength"}
 
     # We expect the whole [0, 360] to be covered
-    assert len(results["lo"].coords["vaa"]) == 360 / 180
+    assert len(results["lo"].coords["vaa"]) == 360. / 180.
     # # We expect [0, 90[ to be covered (90° should be missing)
-    assert len(results["lo"].coords["vza"]) == 90 / 45
+    assert len(results["lo"].coords["vza"]) == 90. / 45.
     # We just check that we record something as expected
-    assert np.all(results["lo"].data > 0)
+    assert np.all(results["lo"].data > 0.)
 
 
 @pytest.mark.slow
 def test_onedim_solver_app_spp_splitting():
-    """Test the spp splitting mechanism by running the same scene, once with
-    single and double precision each. The scene config contains an SPP value
-    high enough for a sensor split up to occur in the single precision mode."""
-    import numpy as np
-
-    config = {
-        "measure": [{
-            "type": "toa_hsphere",
-            "zenith_res": 45.,
-            "azimuth_res": 180.,
-            "spp": 1000000,
-            "hemisphere": "back"
-        }]
-    }
+    """Test the SPP splitting mechanism by running the same scene, once with
+    single and double precision each.
+    """
+    zenith_res = 90.
+    spp = 1000000
+    n_split = 5
 
+    # Double-precision run
     eradiate.set_mode("mono_double")
-    app = OneDimSolverApp(deepcopy(config))
+    app = OneDimSolverApp(scene=OneDimScene(measures=TOAPPlaneMeasure(
+        zenith_res=zenith_res,
+        spp=spp,
+        spp_max_single=spp,
+    )))
     app.run()
-    results_double = app.results["toa_hsphere"]
+    results_double = app.results["toa_pplane"]["lo"].values
 
+    # Single-precision run
     eradiate.set_mode("mono")
-    app = OneDimSolverApp(deepcopy(config))
+    app = OneDimSolverApp(scene=OneDimScene(measures=TOAPPlaneMeasure(
+        zenith_res=zenith_res,
+        spp=spp,
+        spp_max_single=spp,
+    )))
+    app.run()
+    results_single = app.results["toa_pplane"]["lo"].values
+
+    # Single-precision with splitting
+    app = OneDimSolverApp(scene=OneDimScene(measures=TOAPPlaneMeasure(
+        zenith_res=zenith_res,
+        spp=spp,
+        spp_max_single=min(spp / n_split, int(1e5)),
+    )))
     app.run()
-    results_single = app.results["toa_hsphere"]
+    results_single_split = app.results["toa_pplane"]["lo"].values
 
-    assert np.allclose(results_single["lo"], results_double["lo"], rtol=1e-5)
+    deviation_single = (
+            np.abs(results_double - results_single) / results_double
+    ).flatten()
+    deviation_single_split = (
+            np.abs(results_double - results_single_split) / results_double
+    ).flatten()
 
+    assert np.allclose(deviation_single, deviation_single_split, atol=5e-3)
```

Comment:
Isn't your test showing, that the SPP splitting mechanism is not working? Or do we expect the results to be close for 1e6 samples?

Reply:
In order to demonstrate that SPP splitting works as expected, you have to show that it yields results similar to a computation without splitting. There is no perfect way of testing this: situations where double precision is really required are not something like this test case ([unless you plan on shooting 1e8+ rays](https://github.com/mitsuba-renderer/mitsuba2/issues/82)). 

What I did instead is force a split with a lower number of rays. With the values I used (I kept 1e6 for `spp`), the maximum SPP number in the second single-precision computation is 1e5: splitting occurs. But still, we have access, with 1e6 SPP, to an accurate value without splitting. I could arguably add something like this to also check that the single-precision reference is good:
```python
assert np.all(deviation_single < 1e-3)
```
(we still have quite a bit of noise though so the threshold may be inappropriate)

Reply:
This however, indeed, doesn't show that splitting improves anything. For that, we'll need to run a much longer test case, and I don't think we can realistically put this in our automated testing suite.